ml_metadata/tools/mlmd_bench/fill_events_workload.cc

```diff
@@ -0,0 +1,253 @@
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+#include "ml_metadata/tools/mlmd_bench/fill_events_workload.h"
+
+#include <math.h>
+
+#include <random>
+#include <vector>
+
+#include "ml_metadata/metadata_store/metadata_store.h"
+#include "ml_metadata/metadata_store/types.h"
+#include "ml_metadata/proto/metadata_store_service.pb.h"
+#include "ml_metadata/tools/mlmd_bench/proto/mlmd_bench.pb.h"
+#include "ml_metadata/tools/mlmd_bench/util.h"
+#include "tensorflow/core/lib/core/errors.h"
+#include "tensorflow/core/lib/core/status.h"
+#include "tensorflow/core/platform/logging.h"
+
+namespace ml_metadata {
+namespace {
+
+// Generates and returns a categorical distribution with Dirichlet Prior
+// specified by `dist`.
+std::discrete_distribution<int64>
+GenerateCategoricalDistributionWithDirichletPrior(
+    const int64 sample_size, const CategoricalDistribution& dist,
+    std::minstd_rand0& gen) {
+  // With a source of Gamma-distributed random variates, draws `sample_size`
+  // independent random samples and store in `weights`.
+  std::gamma_distribution<double> gamma_distribution(dist.dirichlet_alpha(),
+                                                     1.0);
+  std::vector<double> weights(sample_size);
+  for (int64 i = 0; i < sample_size; ++i) {
+    weights[i] = gamma_distribution(gen);
+  }
+  // Uses these random number generated w.r.t. a Dirichlet distribution with
+  // `concentration_param` to represent the possibility of being chosen for each
+  // integer within [0, sample_size) in a discrete distribution.
+  return std::discrete_distribution<int64>{weights.begin(), weights.end()};
+}
+
+// Generates and returns a zipf distribution with a configurable `skew`
+// specified by `dist`.
+std::discrete_distribution<int64> GenerateZipfDistributionWithConfigurableSkew(
+    const int64 sample_size, const ZipfDistribution& dist) {
+  std::vector<double> weights(sample_size);
+  for (int64 i = 0; i < sample_size; ++i) {
+    // Here, we discard the normalize factor since the `discrete_distribution`
+    // will perform the normalization for us.
+    weights[i] = 1 / pow(i, dist.skew());
+  }
+  // Uses these random number generated w.r.t. a zipf distribution with a
+  // configurable `skew` to represent the possibility of being chosen for each
+  // integer within [0, sample_size) in a discrete distribution.
+  return std::discrete_distribution<int64>{weights.begin(), weights.end()};
```

Comment:
Yes, it looks well.
You can paste the below code in C++ shell to take a look:
`
// discrete_distribution
#include <iostream>
#include <random>
#include <vector>

int main()
{
  const int nrolls = 10000; // number of experiments
  const int nstars = 1000;   // maximum number of stars to distribute

  std::default_random_engine generator;
  int sample_size = 1000;
  std::vector<double> weights(sample_size);
  for (int i = 1; i <= sample_size; ++i) {
    // Here, we discard the normalize factor since the `discrete_distribution`
    // will perform the normalization for us.
    weights[i - 1] = 1 / pow(i, 1);
  }
  // Uses these random number generated w.r.t. a zipf distribution with a
  // configurable `skew` to represent the possibility of being chosen for each
  // integer within [0, sample_size) in a discrete distribution.
  std::discrete_distribution<int> distribution {weights.begin(), weights.end()};

  int p[sample_size]={};

  for (int i=0; i<nrolls; ++i) {
    int number = distribution(generator);
    ++p[number];
  }

  std::cout << "a discrete_distribution:" << std::endl;
  for (int i=0; i<sample_size; ++i)
    std::cout << i << ": " << std::string(p[i]*nstars/nrolls,'*') << std::endl;

  return 0;
}
`